sklearn/utils/_param_validation.py

```diff
@@ -415,7 +415,7 @@ def __str__(self):
         )
 
 
-def generate_invalid_param_val(constraint):
+def generate_invalid_param_val(constraint, constraints=None):
```

Comment:
I don't know if it would be more explicit to have `other_contraints` instead of only `constraints`?

Reply:
it's not the other constraints, its all the constraints, including `constraint`

Reply:
But do we need the current constraint?

Reply:
constraints is the list of all constraints for the parameter. It would just be more complex to try to extract a specific constraint from the list before calling the function.